pandas/tests/indexes/numeric/test_numeric.py

```diff
@@ -278,10 +281,14 @@ def test_is_strictly_monotonic(self):
         assert not index._is_strictly_monotonic_increasing
         assert not index._is_strictly_monotonic_decreasing
 
-    def test_logical_compat(self, simple_index):
+    def test_logical_compat(self, simple_index, request):
         idx = simple_index
-        assert idx.all() == idx.values.all()
-        assert idx.any() == idx.values.any()
+        msg = "cannot perform all with this index type: Index"
+
+        if request.node.callspec.id not in ["int", "int64"]:
```

Comment:
using request.node.callspec.id is a last resort and not at all clear what you are doing.  cant this be expressed in terms of idx.dtype?

Reply:
I think, we don't need to change anything here, this test passes.